jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+
+			index.handler(event, {}, (err, res) => {
+				expect(JSON.parse(err).message).to.equal("Error getting metrics");			
+				sinon.assert.calledOnce(getQueryStub);
+				sinon.assert.calledOnce(getMetricsStub);
+
+				getQueryStub.restore();
+				getMetricsStub.restore();
+			});
+		});
+
+		it('should return metrics when getResponseForHelpPathString is called successfully', () => {
+			const query = {};
+
+			const getQueryStub = sinon.stub(utils, "getQuery").returns(query);
+			const getMetricsStub = sinon.stub(utils, "getMetrics").returns({ metrics: ['code-coverage'] });
+
+			index.handler(event, {}, (err, res) => {
+				expect(res.data.metrics[0].name).to.equal("code-coverage");
+				sinon.assert.calledOnce(getQueryStub);
+				sinon.assert.calledOnce(getMetricsStub);
+
+				getQueryStub.restore();
+				getMetricsStub.restore();
+			});
+		});
+
+		it('should return metrics when getResponseForCodeqPathString is called successfully', () => {
+			const query = {};
+			event.resourcePath = 'codeq';
+			const getQueryStub = sinon.stub(utils, "getQuery").returns(query);
+			const missingFieldsStub = sinon.stub(validation, "validateMissingFields");
+			const fromDateStub = sinon.stub(validation, "validateFromDate").returns('fromdate');
+			const toDateStub = sinon.stub(validation, "validateToDate").returns('todate');
+			const fromAfterToStub = sinon.stub(validation, "validateFromAfterTo").returns(true);
+			const getMetricsStub = sinon.stub(utils, "getMetrics").returns({metrics: ['code-coverage']});
+			const sonarTokenStub = sinon.stub(utils, 'getSonarToken').resolves({base_auth_token: 'base_token'});
+			const jazzTokenStub = sinon.stub(utils, 'getJazzToken').resolves({auth_token: 'auth_token'});
+			const projectBranchStub = sinon.stub(utils, 'getProjectBranch').resolves({branch: 'testBranch'});
+			const codeqReportStub = sinon.stub(utils, 'getCodeqReport').resolves({get_codeq_report: 'codeQReport'});
+			
+			index.handler(event, {}, (err, res) => {
+				expect(res.data).to.equal("codeQReport");
+
+				sinon.assert.calledOnce(getQueryStub);
+				sinon.assert.calledOnce(getMetricsStub);
+				sinon.assert.calledOnce(missingFieldsStub);
+				sinon.assert.calledOnce(fromDateStub);
+				sinon.assert.calledOnce(toDateStub);
+				sinon.assert.calledOnce(fromAfterToStub);
+				sinon.assert.calledOnce(getMetricsStub);
+				sinon.assert.calledOnce(sonarTokenStub);
+				sinon.assert.calledOnce(jazzTokenStub);
+				sinon.assert.calledOnce(projectBranchStub);
+				sinon.assert.calledOnce(codeqReportStub);
+
+				getQueryStub.restore();
+				getMetricsStub.restore();
+				missingFieldsStub.restore();
+				fromDateStub.restore();
+				toDateStub.restore();
+				fromAfterToStub.restore();
+				getMetricsStub.restore();
+				sonarTokenStub.restore();
+				jazzTokenStub.restore();
+				projectBranchStub.restore();
+				codeqReportStub.restore();
+			});
+		});

```

Comment:
We should mimic the scenario for the catch on `line 72` as well as the `if .. else ..` within it.

Reply:
added